### PR TITLE
Add air quality and water level sensor sensor

### PR DESCRIPTION
**Pull request by philipgiuliani**
I still have to add all the necessary characteristics for an AirQualitySensor-Service.

### CODE REVIEW

File: lib/hap/services/air_quality_sensor.ex [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  """
+
+  defstruct air_quality: nil, name: nil, active: nil, fault: nil, tampered: nil, low_battery: nil
+
+  defimpl HAP.ServiceSource do
+    def compile(value) do
+      HAP.Service.ensure_required!(__MODULE__, "air_quality", value.air_quality)
+
+      %HAP.Service{
+        type: "8A",
```
Comment (mtrudel) [inline lib/hap/services/air_quality_sensor.ex:13]: Per p.134 of HAP R2 this is 8D, no?

File: lib/hap/services/air_quality_sensor.ex [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        characteristics: [
+          {HAP.Characteristics.AirQuality, value.air_quality},
+          {HAP.Characteristics.Name, value.name},
+          # TODO: The following optional characteristics still have to be implemented
+          # ”9.71 Ozone Density” (page 191)
+          # ”9.64 Nitrogen Dioxide Density” (page 189)
+          # ”9.106 Sulphur Dioxide Density” (page 221)
+          # ”9.66 PM2.5 Density” (page 190)
+          # ”9.72 PM10 Density” (page 192)
+          # ”9.126 VOC Density” (page 233)
```
Comment (mtrudel) [inline lib/hap/services/air_quality_sensor.ex:23]: Is it possible to complete these characteristics?

**Review by mtrudel — CHANGES_REQUESTED** [review]
@philipgiuliani thanks for the PR! I've made a couple of notes on here for review. Let me know what you think!

File: lib/hap/services/air_quality_sensor.ex (hunk 1 above)
Comment (philipgiuliani) [reply lib/hap/services/air_quality_sensor.ex]: Thanks, this also explained why the Home App didn't display the Sensor correctly 🙃

File: lib/hap/services/air_quality_sensor.ex (hunk 2 above)
Comment (philipgiuliani) [reply lib/hap/services/air_quality_sensor.ex]: Yes, I had it already planned, just wanted to open the PR so you know that this is WIP.

**Comment by philipgiuliani** [comment]
Merge request is ready!

File: lib/hap/characteristics/voc_density.ex [hunk 3]
```diff
@@ -0,0 +1,13 @@
+defmodule HAP.Characteristics.VocDensity do
```
Comment (mtrudel) [inline lib/hap/characteristics/voc_density.ex:1]:
```suggestion
defmodule HAP.Characteristics.VOCDensity do
```

File: lib/hap/services/air_quality_sensor.ex [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        type: "8D",
+        characteristics: [
+          {HAP.Characteristics.AirQuality, value.air_quality},
+          {HAP.Characteristics.Name, value.name},
+          {Hap.Characteristics.OzoneDensity, value.ozone_density},
+          {HAP.Characteristics.NitrogenDioxideDensity, value.nitrogen_dioxide_density},
+          {HAP.Characteristics.SulphurDioxideDensity, value.sulphur_dioxide_density},
+          {HAP.Characteristics.PM25Density, value.pm2_5_density},
+          {HAP.Characteristics.PM10Density, value.pm10_density},
+          {HAP.Characteristics.VocDensity, value.voc_density},
```
Comment (mtrudel) [inline lib/hap/services/air_quality_sensor.ex:33]:
```suggestion
          {HAP.Characteristics.VOCDensity, value.voc_density},
```

**Review by mtrudel — CHANGES_REQUESTED** [review]
@philipgiuliani looks good - I'm just going to change the capitalization on one module name and merge

File: lib/hap/characteristics/voc_density.ex (hunk 3 above)
Comment (mtrudel) [inline lib/hap/characteristics/voc_density.ex:1]:
```suggestion
defmodule HAP.Characteristics.VOCDensity do
```

**Comment by mtrudel** [comment]
Looks like you don't have owner updates enabled on your branch so I was unable to make the change - could you please change the two occurrences of `VocDensity` in code to `VOCDensity`? No need to change filenames; it's correct in both cases.

**Comment by mtrudel** [comment]
Good to go after that!

**Comment by philipgiuliani** [comment]
Done 👍

**Comment by mtrudel** [comment]
Merged; thanks for the PR!

**Comment by mtrudel** [comment]
Released as 0.4.5

**Pull request merged**